data-serving/data-service/api/openapi.yaml

```diff
@@ -94,7 +100,7 @@ paths:
         content:
           application/json:
             schema:
-              $ref: "#/components/schemas/NewCase"
+              $ref: "#/components/schemas/Case"
```

Comment:
Aren't the same values `required` for upsert?

Reply:
AFAICT, from the tests, no -- since it doesn't require a full `case` object -- LMK if that's incorrect though and I can fix the tests

I did add the `caseReference` field as required though since I think that *is* required for upsert

Reply:
You're right on about that -- erroneous comment on my part. The mechanics are a little complicated:

1. Upsert looks for a case by the provided `cR.sourceId` and `cR.sourceEntryId`.
2. If it finds one, it performs an update, potentially partial, with the remainder of the request body.
3. If it doesn't find one, it creates an item with the provided request body; and in this case, `cR` must be present as you'd expect.

The result is that `cR` is effectively required, but depending on whether a case is found, other fields are semantically required (or not).

Reply:
makes sense, thank you!